File: Commons/Materials/clsMaterials.vb
Imports System.Xml

Public Enum MaterialType
    CommonBox        '盒子
    ActiveBox        '盒子（选中）
    BoundingBox      '盒子的包围盒
    GameBoxFrame
    GameBoxBed
    AxisOrgin
    AxisX
    AxisY
    AxisZ
End Enum

Public Class clsMaterials
    Private MaterialConfig As System.Xml.XmlElement  '材料配置集合

    Public Sub New()
        Dim XML As New XmlDocument
        XML.LoadXml(My.Resources.ConfigMaterial)
        Dim node As System.Xml.XmlElement
        node = XML.SelectSingleNode("Materials")
        Me.MaterialConfig = node
    End Sub

    Public Sub SetMaterial(ByVal mt As MaterialType)
        Dim MType As String = ""
        Select Case mt
            Case MaterialType.CommonBox
                MType = "CommonBox"
            Case MaterialType.ActiveBox
                MType = "ActiveBox"
            Case MaterialType.BoundingBox
                MType = "BoundingBox"
            Case MaterialType.GameBoxFrame
                MType = "GameBoxFrame"
            Case MaterialType.GameBoxBed
                MType = "GameBoxBed"
            Case MaterialType.AxisX
                MType = "AxisX"
            Case MaterialType.AxisY
                MType = "AxisY"
            Case MaterialType.AxisZ
                MType = "AxisZ"
            Case MaterialType.AxisOrgin
                MType = "AxisOrgin"
        End Select
        Dim node As System.Xml.XmlElement
        node = MaterialConfig.SelectSingleNode(MType)
        Dim ambient, diffuse, specular, emission, shininess As String
        With node
            ambient = .SelectSingleNode("ambient").InnerXml
            diffuse = .SelectSingleNode("diffuse").InnerXml
            specular = .SelectSingleNode("specular").InnerXml
            emission = .SelectSingleNode("emission").InnerXml
            shininess = .SelectSingleNode("shininess").InnerXml
        End With
        Dim objMT As New clsMaterial(ambient, diffuse, specular, emission, shininess)
        objMT.SetMaterial()
    End Sub

End Class

File: Basic/Materials/clsMaterials.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml;
namespace Commons {
    public enum MaterialType {
        CommonBox,// 盒子
        ActiveBox,//盒子（选中）
        BoundingBox,//盒子的包围盒
        GameBoxFrame,
        GameBoxBed,
        AxisOrgin,
        AxisX,
        AxisY,
        AxisZ
    }
    public class clsMaterials {
        private XmlElement MaterialConfig;

        public clsMaterials() {
            XmlDocument XML = new XmlDocument();
            XML.LoadXml(Commons.Properties.Resources.ConfigMaterial);
            XmlElement node = (XmlElement)XML.SelectSingleNode("Materials");
            MaterialConfig = node;
        }

        public void SetMaterial(MaterialType mt) {
            string MType = "";
            switch (mt) {
                case MaterialType.CommonBox:
                    MType = "CommonBox"; break;
                case MaterialType.ActiveBox:
                    MType = "ActiveBox"; break;
                case MaterialType.BoundingBox:
                    MType = "BoundingBox"; break;
                case MaterialType.GameBoxFrame:
                    MType = "GameBoxFrame"; break;
                case MaterialType.GameBoxBed:
                    MType = "GameBoxBed"; break;
                case MaterialType.AxisX:
                    MType = "AxisX"; break;
                case MaterialType.AxisY:
                    MType = "AxisY"; break;
                case MaterialType.AxisZ:
                    MType = "AxisZ"; break;
                case MaterialType.AxisOrgin:
                    MType = "AxisOrgin"; break;
            }
            XmlElement node = (XmlElement)MaterialConfig.SelectSingleNode(MType);
            string ambient, diffuse, specular, emission, shininess;
            ambient = node.SelectSingleNode("ambient").InnerXml;
            diffuse = node.SelectSingleNode("diffuse").InnerXml;
            specular = node.SelectSingleNode("specular").InnerXml;
            emission = node.SelectSingleNode("emission").InnerXml;
            shininess = node.SelectSingleNode("shininess").InnerXml;

            clsMaterial objMT = new clsMaterial(ambient, diffuse, specular,
                emission, shininess);
            objMT.SetMaterial();
        }

    }
}
